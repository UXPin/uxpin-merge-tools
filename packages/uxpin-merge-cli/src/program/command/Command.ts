export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  DELETE_VERSION = 'delete-version',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND: Command = Command.EXPERIMENT;
