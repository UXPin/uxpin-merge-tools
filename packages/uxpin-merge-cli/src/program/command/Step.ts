import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun: boolean;
  exec: (infos: DSMetadata) => any;
}

export type StepExecutor = (designSystem: DSMetadata) => Promise<DSMetadata>;

export type StepWithoutDSExecutor = () => Promise<void>;
