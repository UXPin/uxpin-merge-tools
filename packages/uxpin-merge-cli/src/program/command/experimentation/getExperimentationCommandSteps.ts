import { Store } from '../../../utils/store/Store';
import { ExperimentProgramArgs } from '../../args/ProgramArgs';
import { Step } from '../Step';
import { experimentationBuildLibraryStep } from './steps/experimentationBuildLibraryStep';
import { experimentationCreateEpidStep } from './steps/experimentationCreateEpidStep';
import { experimentationOpenBrowserStep } from './steps/experimentationOpenBrowserStep';
import { experimentationPrepareMetadataStep } from './steps/experimentationPrepareMetadataStep';
import { experimentationRunNgrok } from './steps/experimentationRunNgrok';
import { experimentationRunServerStep } from './steps/experimentationRunServerStep';

export interface ExperimentationState {
  ngrokUrl?: string;
}

const defaultState: ExperimentationState = {};

export function getExperimentationCommandSteps(args: ExperimentProgramArgs): Step[] {
  const store: Store<ExperimentationState> = new Store(defaultState);

  return [
    experimentationBuildLibraryStep(args),
    experimentationRunNgrok(args, store),
    experimentationCreateEpidStep(args),
    experimentationPrepareMetadataStep(args),
    experimentationRunServerStep(args, store),
    experimentationOpenBrowserStep(args, store),
  ];
}
