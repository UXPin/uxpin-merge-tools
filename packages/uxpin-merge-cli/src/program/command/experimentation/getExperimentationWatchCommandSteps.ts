import { print } from '../../../utils/console/print';
import { printLine } from '../../../utils/console/printLine';
import { PrintColor } from '../../../utils/console/PrintOptions';
import { ExperimentProgramArgs } from '../../args/ProgramArgs';
import { Step } from '../Step';
import { experimentationBuildLibraryStep } from './steps/experimentationBuildLibraryStep';
import { experimentationPrepareMetadataStep } from './steps/experimentationPrepareMetadataStep';

export const COMPILATION_SUCCESS_MESSAGE = '🌐 Compiled successfully! Now you can refresh your browser.';

export function getExperimentationWatchCommandSteps(args: ExperimentProgramArgs): Step[] {
  return [
    onCompilationStart(),
    experimentationBuildLibraryStep(args),
    experimentationPrepareMetadataStep(args),
    onCompilationEnd(),
  ];
}

function onCompilationStart(): Step {
  return {
    exec: () => print('🎩 Hang on to your hat! Compilation in progress...', { color: PrintColor.YELLOW }),
    shouldRun: true,
  };
}

function onCompilationEnd(): Step {
  return {
    exec: () => printLine(`\r${COMPILATION_SUCCESS_MESSAGE}`),
    shouldRun: true,
  };
}
