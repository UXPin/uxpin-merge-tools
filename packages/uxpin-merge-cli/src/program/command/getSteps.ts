import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getDeleteVersionCommandSteps } from './delete_version/getDeleteVersionCommandSteps';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getExperimentationCommandSteps } from './experimentation/getExperimentationCommandSteps';
import { getGeneratePresetsCommandSteps } from './generate_presets/getGeneratePresetsCommandSteps';
import { getInitCommandSteps } from './init/getInitCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args: ProgramArgs): Step[] {
  switch (args.command) {
    case Command.DUMP:
      return getDumpCommandSteps();

    case Command.EXPERIMENT:
      return getExperimentationCommandSteps(args);

    case Command.INIT:
      return getInitCommandSteps(args);

    case Command.GENERATE_PRESETS:
      return getGeneratePresetsCommandSteps(args);

    case Command.PUSH:
      return getPushCommandSteps(args);

    case Command.DELETE_VERSION:
      return getDeleteVersionCommandSteps(args);

    case Command.SERVER:
      return getServerCommandSteps(args);

    case Command.SUMMARY:
      return getSummaryCommandSteps();

    default:
      return [];
  }
}
