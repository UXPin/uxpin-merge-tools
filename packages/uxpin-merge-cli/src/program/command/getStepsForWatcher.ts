import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args: ProgramArgs): Step[] {
  if (args.command === Command.EXPERIMENT) {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
