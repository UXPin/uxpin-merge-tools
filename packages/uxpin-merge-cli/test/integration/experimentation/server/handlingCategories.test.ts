import { AxiosResponse } from 'axios';
import { OK } from 'http-status-codes';
import { setTimeoutBeforeAll } from '../../../utils/command/setTimeoutBeforeAll';
import { setupExperimentationServerTest } from '../../../utils/experimentation/setupExperimentationServerTest';

const CURRENT_TIMEOUT = 300000;
setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('Experimentation mode - handling categories', () => {
  let response: AxiosResponse;
  const { axiosPromise } = setupExperimentationServerTest({
    projectPath: 'resources/designSystems/twoComponentsWithConfig',
    serverCmdArgs: ['--webpack-config "./webpack.config.js"'],
  });

  beforeAll(async () => {
    const origin = 'https://app.uxpin.com';
    response = await axiosPromise('/code/categories', { headers: { origin } });
  });

  it('should responds with OK status code', async () => {
    expect(response.status).toEqual(OK);
  });

  it('should responds with correct CORS headers and no-cache', async () => {
    // given
    const expectedHeaders: any = {
      'access-control-allow-credentials': 'true',
      'access-control-allow-headers': 'Origin, X-Requested-With, Content-Type, Accept, Range',
      'access-control-allow-origin': 'https://app.uxpin.com',
      'cache-control': 'no-cache',
      'content-type': 'application/json',
    };

    // then
    expect(response.headers).toEqual(expect.objectContaining(expectedHeaders));
  });

  it('should respond with proper body', () => {
    expect(response.data).toMatchSnapshot();
  });
});
