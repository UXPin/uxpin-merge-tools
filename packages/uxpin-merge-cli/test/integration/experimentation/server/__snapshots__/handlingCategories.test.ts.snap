// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experimentation mode - handling categories should respond with proper body 1`] = `
Array [
  Object {
    "id": 1,
    "isDefault": true,
    "name": "General",
    "sortIndex": 0,
    "type": "code-sync-component",
  },
  Object {
    "id": 2,
    "isDefault": false,
    "name": "Forms",
    "sortIndex": 1,
    "type": "code-sync-component",
  },
]
`;
